src/UsblParser.hpp

```diff
@@ -3,114 +3,43 @@
-             * The function interpretes the response as OK.
-             * If the respnse isn't an OK the function throws an Exception.
-             * @param[in] data The data in a uint8 array to analysing
-             * @param[in] size The length of data
-             */
-            static void parseOk(uint8_t const* data, size_t const size);
-            /*
-             * The function interpretes the string as a connection status and
-             * returns a enum with the status.
-             *
-             * @param[in] command The command to analysing
-             * @return The connection status in a enum
-             */
-            static ConnectionStatus parseConnectionStatus(std::string const command);
-            /*
-             * The function interpretes the string as a incoming instant message and returns
-             * the extracted instant message
-             *
-             * @param[in] command The command to analysing
-             * @return The instant message in a struct
-             */
-            static ReceiveInstantMessage parseIncomingIm(std::string const s);
-            /*
-             * The functions interpretes the string as a position and
-             * returns a struct with the position.
-             *
-             * @param[in] command The command to analysing
-             * @return The position in a struct
-             */
-            static Position parsePosition(std::string const command);
-            static Position parseRemotePosition(std::string const s);
-            /*
-             * The funcion extracts the string in the response and returns the string.
-             * You have to interprete the string anymore.
-             * The function throws an exception, if the response isn't fit to the command.
-             *
-             * @param[in] data The data in a uint8 array to analysing
-             * @param[in] size The length of data
-             * @param[in] command The command to fit the data with this command 
-             * @return The extracted string
-             */
-            static std::string parseString(uint8_t const* data, size_t const size, std::string const command);
+            ~UsblParser();
+
+//            int isPacket(std::string const &buffer, OperationMode const &mode);
+
+            bool findNotification(std::string const &buffer, Notification &notification);
+            bool validateNotification(std::string const &buffer);
+            bool splitValidateNotification(std::string const &buffer, Notification const &notification, std::string &output_msg);
+
+
+            bool findResponse(std::string const &buffer, CommandResponse &response);
+            bool validateResponse(std::string const &buffer, std::string const &command);
+            bool validateParticularResponse(std::string const &buffer);
+
+
+            std::string parseSendIM(SendIM const &im);
+
+            std::string parseReceivedIM(std::string const &buffer, ReceiveIM &im);
```

Comment:
You fill in a structure *and* return a string ?

I think you will definitely have to document all that stuff ...